### ISSUE TITLE
Deprecate the support for backup without VolumeInfo metadata

### ISSUE THREAD

**Issue by blackpiglet** [vmware-tanzu/velero#8076]
**Describe the problem/challenge you have**

In the v1.13 release note's `Deprecation Announcement` section: https://github.com/vmware-tanzu/velero/releases/v1.13.0/, support for the backup without the VolumeInfo metadata file is announced deprecated, which means not supporting restore the backup without VolumeInfo metadata.

After removing this support for the Backup without VolumeInfo metadata, the legacy backup's PV data cannot be restored.

This piece of code will be deleted to deprecate the support.
https://github.com/vmware-tanzu/velero/blob/d4e743b138be03d7491a7408058580d2c8ae71aa/pkg/restore/restore.go#L1246-L1295

Please check whether it's OK to make the change in v1.15.
@shubham-pampattiwar @sseago @Lyndon-Li @reasonerjt @anshulahuja98 @kaovilai 

**Describe the solution you'd like**

**Anything else you would like to add:**

**Environment:**

- Velero version (use `velero version`):
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

**Comment by blackpiglet** [comment]
No, I'm OK with keeping the existing code.

**Comment by sseago** [comment]
> Hi I probably didn't see this note during 1.13 release. I believe this is the first time we are breaking restore support from older backups in velero.
> 
> Handling this migration is non-trivial for any user, because there is no straightline path to recovery. Even for downstream, it becomes very hard to do this migration.
> 
> Do you see major complications in keeping back compat for longer / or is it blocking other features?

@anshulahuja98 Note that according to the deprecation policy, something deprecated in release N cannot be removed until release N+2 -- however, any feature required for restore must allow backups from a given release to be restorable for 2 more releases. This means that if a feature is deprecated in 1.13, it's removed (in general) in 1.15, but any backup from 1.14 that uses deprecated functionality must still be restorable in 1.16.

In other words, deprecate in 1.13, remove feature from new backups in 1.15, but backups relying on deprecated functionality will continue to work until 1.17, where the feature is removed completely.

https://github.com/vmware-tanzu/velero/blob/main/GOVERNANCE.md#deprecation-window

**Comment by anshulahuja98** [comment]
I understand the scenario you explained Scott.

But If I correctly understand this particular issue, here the removal/ change to backupVolumeInfo was done in 1.13 and from 1.15 restores will not work

Here there is no explicit removal from backup so 1.13-> 1.15 -> 1.17 is not really something which will hold

Overall based on my understanding we moved to BackupVolumeInfo in 1.13 and we are blocking restores from <=1.12 in 1.15

Given this is a relatively small backcompat to handle, I would prefer to keep restorability for longer. Unless it is stopping us from building new features OR has complications with maintaining over time (for ex Restic).

**Comment by sseago** [comment]
I think part of the confusion here is that the the deprecation policy is trying to communicate a couple different things, and it may be easy to conflate them. First is when we deprecate something, we cannot remove it for a minimum of 2 releases. Second, a backup that makes use of a deprecated feature must also be restoreable for a minimum of 2 releases. With something like restic, this is clear enough -- if we deprecate in release N, a user creates a restic backup in N+1, cannot create one in N+2, but that N+1 restic backup must still be restorable in N+3, but not in N+4.

With VolumeInfo it's a bit less obvious -- but it seems that if any backup from 1.12 can be restored in 1.14, and an backup from 1.13 can be restored in 1.15, then we are following the policy.

Note that the policy is a statement of minimum support. If you want to make the case that this feature in particular needs support to restore a backup more than 2 releases old, then that is one possibility. If you want to discuss the policy itself -- i.e. "I think that a backup from any particular release of velero should be restorable for 4 releases not just 2" -- that's another possibility. I guess we have 3 options here:

1) 2 releases is enough for this feature. It's fine for 1.12 backups to fail restore in 1.15.
2) This feature is different (because it affects more users, all users, etc.) -- so for this feature in particular we should allow 1.12 backups to be restorable for longer than 2 releases. So if we don't remove in 1.15, then when? 1.16? 1.17?
3) We need to rethink how long backups should be valid. We really want any backup from Velero version N to be fully resorable on a version later than N+2. How much to extend?

It might be good to think about calendar time here instead of number of releases. The scenario here, in particular, is "a backup taken the day before upgrading to release N may stop working when later upgrading to release N+2".
If we assume that you upgrade to each new release on some consistent number of days post-release -- say 2 weeks after release, or 4 -- it doesn't really matter to the point here as long as it's consistent) -- this means that you can only assume that a backup is as good as however long it takes for Velero to release 2 minor versions. Looking at 1.12 to 1.14 time frame, that was about 9 months. If we bumped that up to 3 releases for "old backup restore works" instead of 2, then we'd have a little over a year's lifetime for a backup.

Any thoughts?

**Comment by anshulahuja98** [comment]
Yes, I agree with your point that we should probably think of this in terms of timelines. 

One important thing to callout here is the retention a customer might have. If a customer is looking for retaining for longer duration on cheaper storage (eg Cold Tier of S3)

In our downstream consumption we intend to offer to our customers potentially very long-term retention (upto 10years) for audit purposes. While I understand 10 years might be very big number to target here. Consider this out of scope of current discussion.

But something like a 2-year period is what seems reasonable for me.

I also want to bring here the nuance that each feature might have a different story, What I mean by that is that, there might be scenarios such as VolumeInfo where even if we don't deprecate in n+2, it is not fundamentally blocking any new feature development by keeping backup compat. And to the extent I understand maintaining this piece of code will not have lot of overhead.

Whereas something like restic will continue to demand our attention, either through customer created issues/ bumping versions/ CVEs, and might have bigger overhead and it might have a stronger case for deprecation.

Overall, my suggestion based on the thoughts I shared above is that we need to clarify deprecation policy to avoid confusion and add more nuance perhaps.

For features such as restic where we slowly phase out backup and then phase out restore ||  n-2 -> Start showing Warning ->n-> Stop Backup support -> n+2 -> Stop Restore support ->

For features such as VolumeInfo where we stop backup support in say N and there is no scenario for showing warning in N-2, we can consider deprecating in N+4

Just writing it based on how I understand, we can think through how to write it properly.
